primedev/core/filesystem/filesystem.cpp

```diff
@@ -10,23 +10,23 @@ std::string sCurrentModPath;
 
 ConVar* Cvar_ns_fs_log_reads;
 
-SourceInterface<IFileSystem>* g_pFilesystem;
+IFileSystem* g_pFilesystem;
 
 std::string ReadVPKFile(const char* path)
 {
 	// read scripts.rson file, todo: check if this can be overwritten
-	FileHandle_t fileHandle = (*g_pFilesystem)->m_vtable2->Open(&(*g_pFilesystem)->m_vtable2, path, "rb", "GAME", 0);
+	FileHandle_t fileHandle = g_pFilesystem->m_vtable2->Open(&g_pFilesystem->m_vtable2, path, "rb", "GAME", 0);
 
 	std::stringstream fileStream;
 	int bytesRead = 0;
 	char data[4096];
 	do
 	{
-		bytesRead = (*g_pFilesystem)->m_vtable2->Read(&(*g_pFilesystem)->m_vtable2, data, (int)std::size(data), fileHandle);
+		bytesRead = g_pFilesystem->m_vtable2->Read(&g_pFilesystem->m_vtable2, data, (int)std::size(data), fileHandle);
 		fileStream.write(data, bytesRead);
 	} while (bytesRead == std::size(data));
 
-	(*g_pFilesystem)->m_vtable2->Close(*g_pFilesystem, fileHandle);
+	g_pFilesystem->m_vtable2->Close(&*g_pFilesystem, fileHandle);
```

Comment:
```suggestion
	g_pFilesystem->m_vtable2->Close(g_pFilesystem, fileHandle);
```
these should cancel out